design-reference/architecture/VersionAndGo/readme.md

```diff
@@ -0,0 +1,113 @@
+
+```
+infra/
+ |_ Application.tf
+ |_ Application.yaml
+```
+
+These files should be derived from the template folder in dev repo and cloned into the eng repo. Proposed folder path/location is:
+
+```
+infra/
+  |_template
+      |_template_name
+          |_tests
+          |_config
+          |   |_ application.tf
+          |   |_ application.yml
+          |_template files...
+```
+
+ - `Appliation.tf` is the artifact that points to the template in the eng repo. The application team would need to fill this file out with the requested configuration. Example:
+
+ ```
+ module "az-isolated-service-single-region" {
+   source = "git://dev.azure.com/project/repo?ver=1.0.0//infra/template/az-isolated-service-single-region/"
+   resource_group_location = "eastus"
+   name = "az-simple"
+   deployment_targets = [{
+     app_name = "cobalt-backend-api",
+     image_name = "msftcse/az-service-single-region",
+     image_release_tag_prefix = "release"
+   }]
+   acr_build_git_source_url = "https://github.com/erikschlegel/echo-server.git"
+}
+ ```
+
+ - `Application.yaml` - is the artifact that points to the azure-pipeline.yml file in the eng repo that contains the build/test/deploy pipeline.
+
+This folder/files should be checked into the root of the app repo. The Eng team would then configure a new pipeline in the app teams Azure Devops project to point to these files and deploy the requested infrastructure.
+
+> Note: We should strive to continue to use a single set of pipeline files for all pipelines referenced in this document.
+
+## Version and Go Eng Pipeline Description
+
+This pipeline will produce an artifact that is stored in a location that would be accessible by both the engineering team and the app team. It should follow or use the same pipeline as the development team and app team pipelines. This code that publishes a new version could be controlled by a feature flag as global variable in the eng pipeline. This could also use a SemVer global variable and only create a new artifact when this has been incremented. Would could also leave this as a "manual" process. Manual only means that the pipeline doesn't create the artifact. We could use a script or the likes to do this.
+
+## Version and Go App Pipeline Description
+
+The pipeline "bootstrap" file: `Application.yaml` should perform the following...
+
+1. Request to copy the pipeline files down to the build agent from the engineering repo.
```

Comment:
is this a source code checkout? Unclear what "pipeline files" are. Scripts used by the templates? If so, i'm interested to learn how AzDO handles this

Reply:
The idea is we use a "copy" task to download azure-pipelines.yaml and dependant files from the Operations repo down to the build agent and execute them similar to how we use the build and release templates now.